### DISCUSSION TITLE
Middleware to connect database and inject into Prisma Client

### DISCUSSION THREAD

**Discussion by Mananshi** [prisma/prisma#15054] [Q&A]
I am trying to implement a multi-tenant architecture in my Express Application, and I wanted to write middleware to connect to a database and inject it into the Prisma client dynamically. Can someone guide me as to how that can be done?
So far, the resources I've obtained are an npm package [prisma-multi-tenant](https://www.npmjs.com/package/prisma-multi-tenant) and a few blogs
https://medium.zenika.com/building-a-multi-tenant-application-with-prisma-11bf890304d6
https://darioielardi.dev/schema-based-multi-tenancy-with-nestjs-and-prisma (I'd like to use Node instead of Nestjs)

**Comment by austincrim** [comment]
Hey @Mananshi 👋,

A single instance of the `PrismaClient` can only connect to a single database simultaneously. You could potentially write some logic to create new instances of `PrismaClient` per database, but that would not live a Prisma middleware since they are attached per `PrismaClient` instance.

You can provide a connection string to Prisma at runtime using the [`datasources` parameter](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#datasources) in the `PrismaClient` constructor.

```ts
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient({
  datasources: {
    db: {
      url: 'some connection string',
    },
  },
})
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
